src/native-stack/navigators/createNativeStackNavigator.tsx

```diff
@@ -1,33 +1,33 @@
-import * as React from 'react';
 import {
   createNavigatorFactory,
-  useNavigationBuilder,
   EventArg,
-  StackRouter,
+  StackActions,
   StackNavigationState,
+  StackRouter,
   StackRouterOptions,
-  StackActions,
+  useNavigationBuilder,
 } from '@react-navigation/native';
-
-import {
-  screensEnabled,
-  // eslint-disable-next-line import/no-unresolved
-} from 'react-native-screens';
-import NativeStackView from '../views/NativeStackView';
+import * as React from 'react';
+import { screensEnabled } from 'react-native-screens';
 import {
-  NativeStackNavigatorProps,
-  NativeStackNavigationOptions,
   NativeStackNavigationEventMap,
+  NativeStackNavigationOptions,
+  NativeStackNavigatorProps,
 } from '../types';
+import NativeStackView from '../views/NativeStackView';
 
-function NativeStackNavigator(props: NativeStackNavigatorProps) {
+const NativeStackNavigator: React.FC<NativeStackNavigatorProps> = ({
```

Comment:
Why change it to const?

Reply:
Simply to set the type to `React.FC<NativeStackNavigatorProps>` rather than setting the props type on the argument directly. Fine to change this back if there's seen to be no benefit.

Reply:
Please keep it a function. The reasoning is the same, it's similar to how it is done in `react-navigation`.